internal/api/sdsubmit_test.go

```diff
@@ -156,23 +152,50 @@ func Test_SDSubmit_GetFiles_Pass(t *testing.T) {
 	}
 }
 
-func Test_SDSubmit_GetEnvs_Fail_AccessToken(t *testing.T) {
+func Test_SDSubmit_GetFiles_Split_Pass(t *testing.T) {
 	// Mock
-	expectedError := constantError
-	origGetEnv := getEnv
-	defer func() { getEnv = origGetEnv }()
-	getEnv = func(name string, verifyURL bool) (string, error) {
-		return "", errors.New(expectedError)
+	testFile := []file{
+		{
+			FileID:                "file1",
+			DatasetID:             "dataset1",
+			DisplayFileName:       "file1.txt",
+			FileName:              "file1.txt",
+			FileSize:              20,
+			DecryptedFileSize:     20,
+			DecryptedFileChecksum: "if6ox",
+			FileStatus:            "READY",
+		},
+	}
+	testFileJSON, _ := json.Marshal(testFile)
+	origMakeRequest := makeRequest
+	defer func() { makeRequest = origMakeRequest }()
+	makeRequest = func(url string, query, headers map[string]string, ret interface{}) error {
+		path := "url/metadata/datasets/dataset1/files"
+		if url != path {
+			return fmt.Errorf("makeRequest() received incorrect url\nExpected=%s\nReceived=%s", path, url)
+		}
+		if scheme, ok := query["scheme"]; !ok || scheme != "https" {
+			return fmt.Errorf("makeRequest() received incorrect scheme in query\nExpected=%s\nReceived=%s", "https", scheme)
+		}
+		_ = json.NewDecoder(bytes.NewReader(testFileJSON)).Decode(ret)
+		return nil
 	}
-	s := sdSubmitInfo{token: constantToken}
 
 	// Test
-	err := s.getEnvs()
+	s := submitter{fileIDs: make(map[string]string)}
+	meta, err := s.getFiles("fspath", "url", "https://dataset1")
 
 	if err != nil {
-		if err.Error() != expectedError {
-			t.Errorf("Function failed\nExpected=%s\nReceived=%s", expectedError, err.Error())
-		}
+		t.Fatalf("Function failed, expected no error, received=%v", err)
+	}
+	if len(meta) != 1 {
+		t.Fatalf("Function failed, expected=%d, received=%d", 1, len(meta))
+	}
+	if meta[0].Name != testFile[0].DisplayFileName {
+		t.Errorf("Function failed, expected=%s, received=%s", testFile[0].DisplayFileName, meta[0].Name)
+	}
+	if s.fileIDs["https://dataset1_file1.txt"] != "file1" {
+		t.Errorf("Function failed, expected=%s, received=%s", "file1", s.fileIDs["dataset1_file1.txt"])
```

Comment:
```suggestion
		t.Errorf("Function failed, expected=%s, received=%s", "file1", s.fileIDs["https://dataset1_file1.txt"])
```
Update error message key name also